packages/service-library/src/servicelib/long_running_tasks/_task.py

```diff
@@ -292,17 +304,17 @@ def start_task(
     task_name = f"{handler_module_name}.{handler.__name__}"
 
     # only one unique task can be running
-    if unique and task_manager.is_task_name_running(task_name):
-        managed_tasks_ids = list(task_manager.tasks[task_name].keys())
+    if unique and tasks_manager.is_task_running(task_name):
+        managed_tasks_ids = list(tasks_manager.get_task_group(task_name).keys())
         assert len(managed_tasks_ids) == 1  # nosec
-        managed_task = task_manager.tasks[task_name][managed_tasks_ids[0]]
+        managed_task = tasks_manager.get_task_group(task_name)[managed_tasks_ids[0]]
         raise TaskAlreadyRunningError(task_name=task_name, managed_task=managed_task)
 
     task_progress = TaskProgress.create()
```

Comment:
this create I'm not sure why that is needed.

Reply:
me neither ... I would do it with defaults. I will also change it and let @GitHK comment on it in case we are missing something

Reply:
I updated how ``TaskProgress`` is initialized and updated

Reply:
@sanderegg @GitHK I had to revert the changes I did in progress. 

The way task progress is updated (i.e.  ``TaskProgress``) relies on a keeping a reference to some progress object. This is not guaranteed throughout the code which IMO makes the whole code very unreliable.

Reply:
If this create is only used here, we can remove it and put in the defaults.

@pcrespov not sure what you refer to about the guarantees. But it is correct an instance of `TaskProgress` is injected in the task which is started.
This is used by the status and result callbacks to pull information about the task's progress and send it to the caller.